tests/meta/upload/scout/test_meta_upload_scoutapi_rna.py

```diff
@@ -254,3 +254,173 @@ def test_upload_splice_junctions_bed_to_scout_no_subject_id(
     # THEN an exception should be raised on unconnected data
     with pytest.raises(CgDataError):
         upload_scout_api.upload_splice_junctions_bed_to_scout(case_id=rna_case_id, dry_run=True)
+
+
+def test_upload_rna_fusion_report_to_scout_tumour_non_matching(
+    caplog: Generator[LogCaptureFixture, None, None],
+    dna_case_id: str,
+    mip_rna_analysis_hk_api: HousekeeperAPI,
+    rna_case_id: str,
+    rna_store: Store,
+    upload_scout_api: UploadScoutAPI,
+):
+    """Test that an RNA case's gene fusion report is not uploaded if the is_tumour is not matching"""
+
+    # GIVEN a sample in the RNA case is NOT connected to a sample in the DNA case via is_tumour (i.e. different is_tumour)
+    for link in rna_store.family(rna_case_id).links:
+        link.sample.is_tumour = True
+    for link in rna_store.family(dna_case_id).links:
+        link.sample.is_tumour = False
```

Comment:
You are repeating this block many times - I think it would be great if it was a function

Reply:
moved to a function